File: single-language-analysis/dataset-files/python/test_smell_077.py
import pytest

class TestJmstest:
    def test_smell_077(self):
        self.application_context = TestUtils.instance_context(MODULE, CONFIG_SESSION)
        producer_service = self.application_context.get_bean(ProducerService, "producerService")
        session_aware_queue = self.application_context.get_bean("sessionAwareQueue", Destination)

        producer_service.send_message(session_aware_queue, "TestSessionAwareMessageListener")

        try:
            time.sleep(1)
        except InterruptedException as e:
            print(e)

        print("Instantiating a message listener container, starting to receive messages returned to the producer...")

        container = DefaultMessageListenerContainer()
        container.set_connection_factory(self.application_context.get_bean("connectionFactory", ConnectionFactory))
        container.set_destination(self.application_context.get_bean("queue", Destination))
        container.set_message_listener(ConsumerMessageListener())
        container.initialize()
        container.start()

File: single-language-analysis/dataset-files/java/TestSmell077.java
// Original URL: https://github.com/lijtforgithub/study-spring/blob/a9016a4b8be9202a7b3071879fea719c2531136e/src/main/resources/old/activemq/JMSTest.java#L52-L74

public class TestSmell077 {
    @Test
    public void testSessionAwareMessageListener() {
    	this.applicationContext = TestUtils.instanceContext(MODULE, CONFIG_SESSION);
    	ProducerService producerService = this.applicationContext.getBean(ProducerService.class, "producerService");
    	Destination sessionAwareQueue = this.applicationContext.getBean("sessionAwareQueue", Destination.class);
    	
        producerService.sendMessage(sessionAwareQueue, "测试SessionAwareMessageListener");
        
        try {
			Thread.sleep(1000);
		} catch (InterruptedException e) {
			e.printStackTrace();
		}
        
        System.err.println("实例化一个消息监听容器，开始接收返回给生产者的消息...");
        
        DefaultMessageListenerContainer continer = new DefaultMessageListenerContainer();
        continer.setConnectionFactory(this.applicationContext.getBean("connectionFactory", ConnectionFactory.class));
        continer.setDestination(this.applicationContext.getBean("queue", Destination.class));
        continer.setMessageListener(new ConsumerMessageListener());
        continer.initialize();
        continer.start();
    }
}
